### ISSUE TITLE
Catch-all catches all - need catch-all per domain

### ISSUE THREAD

**Issue by kwijbrans** [docker-mailserver/docker-mailserver#2322]
# Subject

I would like some feedback concerning a use case: catchall account per domain which maps to a virtual mailbox in each domain.

## Description
Hosting multiple domains on a docker-mailserver. Trying to configure a catchall account per domain. This does not seem to work because it catches also the existing mailboxes.

Closed issue: Catch-all catches all, also from existing mailboxes #1454 does not solve it because it used the sedcond answer which assumes there is only one catchall.

I tried to edit what is outlined in the first answer by manually editing /etc/postfix/virtual on the container. When testing this with an e-mail, it seems to work: an e-mail sent from my Outlook account with Microsoft with both me@example.com and test@example.com is correctly put in both the me@domain.com mailbox and the admin@example.com mailbox.

An e-mail only sent to me@example.com is put in the me@example.com mailbox.
An e-mail only sent to test@example.com is put in the admin@example.com mailbox.

``` 
root@mail:/etc/postfix# cat virtual
me@example.com  me@example.com 
admin@example.com  admin@example.com 
@example.com        admin@example.com
me@not-example.com  me@not-example.com 
admin@not-example.com  admin@not-example.com 
@example.com        admin@not-example.com
```
However, how can I configure this using the setup.sh tool only editing the dms/config files?

**Comment by maietta** [comment]
You should probably redact any private information, should you care

**Comment by georglauterbach** [comment]
Not sure whether this is possible, but

``` TXT
me@example.com  me@example.com 
admin@example.com  admin@example.com 
```

does not really make sense to begin with. The `virtual` file is for aliases, and setting `me@example.com` as an alias for `me@example.com` does not make sense. The account already exists and is set up.

To be honest, I', not quite sure whether this works for multiple domains. Maybe someone else from the community can chime in.

**Comment by kwijbrans** [comment]
@georglauterbach, thanks for responding. I googled a bit more and this seems certainly possible. Unfortunately the postfix documentation is not very clear. However, the website http://workaround.org and then especially the page 'Making Postfix get its information from the MariaDB database' gives a more readable explanation of the mechanism that gives me confidence that this will work. Whether the virtual file is used or MariaDB should not really make a difference for the principle.

Found the script that at startup creates the virtual file. However, also found a simple workaround:
- create a virtual file manually as shown above.
- mount this virtual file read only in the docker container to the location where docker mail server expects the virtual file.
- do the same for the vhosts file
Add these files as ro in the volumes of the container:
```  
  volumes:
      - /opt/docker/dms/virtual:/etc/postfix/virtual:ro                 # prevent overwriting by container
      - /opt/docker/dms/vhost:/etc/postfix/vhost:ro
 ```
For the small number of domains and mailboxes that I have, the workaround is fine. When scaling up, people probably will use a SQL DB and a tool like postfixadmin.
